jni/src/test/java/com/horizen/merkletreenative/MerkleTreeTest.java

```diff
@@ -1,314 +1,314 @@
+//        BigLazyMerkleTree smtLazy = BigLazyMerkleTree.init(height, "./db_big_lazy");
+//
+//        //Check leaves position also for BigLazyMerkleTree
+//        i = 0;
+//        for (FieldElement leaf: leaves) {
+//            long position = smtLazy.getPosition(leaf);
+//            assertEquals("Computed position for leaf " + i + " is not the expected one", positions[i], position);
+//            assertEquals("Positions must be the same", position, BigLazyMerkleTree.getPosition(leaf, height));
+//            assertTrue("Position must be empty", smtLazy.isPositionEmpty(position));
+//            i++;
+//        }
+//
+//        //Add leaves to BigLazyMerkleTree
+//        smtLazy.addLeaves(leaves);
+//
+//        //Remove leaves from BigLazyMerkleTree
+//        long[] leavesToRemove = { 458L, 532L };
+//        smtLazy.removeLeaves(leavesToRemove);
+//
+//        //Compute root and assert equality with the expected one
+//        FieldElement smtLazyRoot = smtLazy.root();
+//        assertEquals("BigLazyMerkleTree root is not as expected", smtLazyRoot, expectedRoot);
+//
+//        //Free memory
+//        smtLazy.freeAndDestroyLazyMerkleTree();
+//        smtLazyRoot.freeFieldElement();
+//
+//        //Get InMemoryOptimizedMerkleTree
+//        InMemoryOptimizedMerkleTree mht = InMemoryOptimizedMerkleTree.init(height, numLeaves);
+//
+//        // Must place the leaves at the same positions of the previous trees
+//        List<FieldElement> mhtLeaves = new ArrayList<>();
+//        //Initialize all leaves to zero
+//        FieldElement zero = FieldElement.createFromLong(0L);
+//        for(int j = 0; j < 1024; j++)
+//            mhtLeaves.add(zero);
+//        //Substitute at positions the correct leaves
+//        for (int j = 1; j < numLeaves - 1; j++) {
+//            // Warning: Conversion from long to int is not to be used for production.
+//            mhtLeaves.set((int)positions[j], leaves.get(j));
+//        }
+//
+//        //Append all the leaves to mht
+//        for (FieldElement leaf: mhtLeaves)
+//            mht.append(leaf);
+//
+//        //Finalize the tree
+//        mht.finalizeTreeInPlace();
+//
+//        //Compute root and assert equality with the expected one
+//        FieldElement mhtRoot = mht.root();
+//        assertEquals("InMemoryOptimizedMerkleTree root is not as expected", mhtRoot, expectedRoot);
+//
+//        //It is the same with finalizeTree()
+//        InMemoryOptimizedMerkleTree mhtCopy = mht.finalizeTree();
+//        FieldElement mhtRootCopy = mhtCopy.root();
+//        assertEquals("InMemoryOptimizedMerkleTree copy root is not as expected", mhtRootCopy, expectedRoot);
+//
+//        //Free memory
+//        zero.freeFieldElement();
+//        mht.freeInMemoryOptimizedMerkleTree();
+//        mhtCopy.freeInMemoryOptimizedMerkleTree();
+//        mhtRoot.freeFieldElement();
+//        mhtRootCopy.freeFieldElement();
+//    }
+//
+//    @Test
+//    public void testMerklePaths() {
```

Comment:
Don't see any usage of hardcoded data here. Maybe we can keep it ?

Reply:
This test uses expectedRootBytes hardcoded array